mantidproject/mantid
# Convert table to matrix workspace bug with numeric values

Sometimes the ConvertTableToMatrixWorkspace was failing due to a string, however inspection of the elements revealed that all of the numbers were numeric. A check has been added to see if a string is passed and it then attempts to convert it to a double. Unit tests have also been added to check this scenario and if it will error when a non-numeric string is given. 

**To test:**
1. Run the AlgorithmsTest_ConvertTableToMatrixWorkspaceTest and make sure it passes. 

2. Load a file (e.g. HIFI0008447) and create a table. Then use ConvertTableToMatrixWorkspace on a few different columns (run number usually causes an error prior to the fix). See if the workspace is produced. 


Fixes https://github.com/mantidproject/mantid/issues/18782



---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
